src/test/unit/sensor_gyro_unittest.cc

```diff
@@ -0,0 +1,149 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#include <limits.h>
+#include <algorithm>
+
+extern "C" {
+    #include <platform.h>
+
+    #include "build/build_config.h"
+    #include "build/debug.h"
+    #include "common/axis.h"
+    #include "common/maths.h"
+    #include "common/utils.h"
+    #include "drivers/accgyro/accgyro_fake.h"
+    #include "drivers/accgyro/accgyro_mpu.h"
+    #include "drivers/sensor.h"
+    #include "io/beeper.h"
+    #include "scheduler/scheduler.h"
+    #include "sensors/gyro.h"
+    #include "sensors/acceleration.h"
+    #include "sensors/sensors.h"
+
+    gyroDev_t gyroDev0;
+
+    STATIC_UNIT_TESTED gyroSensor_e gyroDetect(gyroDev_t *dev);
+    struct gyroSensor_s;
+    STATIC_UNIT_TESTED void performGyroCalibration(struct gyroSensor_s *gyroSensor, uint8_t gyroMovementCalibrationThreshold);
+}
+
+#include "unittest_macros.h"
+#include "gtest/gtest.h"
+extern gyroSensor_s * const gyroSensorPtr;
+
+TEST(SensorGyro, Detect)
+{
+    const gyroSensor_e detected = gyroDetect(&gyroDev0);
+    EXPECT_EQ(GYRO_FAKE, detected);
+    EXPECT_EQ(GYRO_FAKE, detectedSensors[SENSOR_INDEX_GYRO]);
+}
+
+TEST(SensorGyro, Init)
+{
+    pgResetAll();
+    const bool initialised = gyroInit();
+    EXPECT_EQ(true, initialised);
+    EXPECT_EQ(GYRO_FAKE, detectedSensors[SENSOR_INDEX_GYRO]);
+}
+
+TEST(SensorGyro, Read)
+{
+    pgResetAll();
+    gyroInit();
+    fakeGyroSet(&gyroDev0, 5, 6, 7);
+    const bool read = gyroDev0.readFn(&gyroDev0);
+    EXPECT_EQ(true, read);
+    EXPECT_EQ(5, gyroDev0.gyroADCRaw[X]);
+    EXPECT_EQ(6, gyroDev0.gyroADCRaw[Y]);
+    EXPECT_EQ(7, gyroDev0.gyroADCRaw[Z]);
+}
+
+TEST(SensorGyro, Calibrate)
+{
+    pgResetAll();
+    gyroInit();
+    printf("c2\r\n");
+    fakeGyroSet(&gyroDev0,5, 6, 7);
+    const bool read = gyroDev0.readFn(&gyroDev0);
+    EXPECT_EQ(true, read);
+    EXPECT_EQ(5, gyroDev0.gyroADCRaw[X]);
+    EXPECT_EQ(6, gyroDev0.gyroADCRaw[Y]);
+    EXPECT_EQ(7, gyroDev0.gyroADCRaw[Z]);
+    static const int gyroMovementCalibrationThreshold = 32;
+    gyroDev0.gyroZero[X] = 8;
+    gyroDev0.gyroZero[Y] = 9;
+    gyroDev0.gyroZero[Z] = 10;
+    performGyroCalibration(gyroSensorPtr, gyroMovementCalibrationThreshold);
+/*!!    EXPECT_EQ(0, gyroDev0.gyroZero[X]);
```

Comment:
What is the reason for having all of these commented out?

Reply:
Fixed.